modules/lang-mustache/src/test/java/org/elasticsearch/messy/tests/TemplateQueryParserTests.java

```diff
@@ -213,8 +214,23 @@ public void testParserCanExtractTemplateNames() throws Exception {
         context.reset(templateSourceParser);
         templateSourceParser.nextToken();
 
+
         TemplateQueryParser parser = injector.getInstance(TemplateQueryParser.class);
-        Query query = parser.fromXContent(context.parseContext()).toQuery(context);
+        Query query = QueryBuilder.rewriteQuery(parser.fromXContent(context.parseContext()), context).toQuery(context);
         assertTrue("Parsing template query failed.", query instanceof MatchAllDocsQuery);
     }
+
+    public void testMustRewrite() throws Exception {
```

Comment:
how about adding this type of test to all the queries where we throw exception as part of toQuery if rewrite wasn't called? or should we in general enforce that rewrite must be always called before calling toQuery?